keras_rs/src/layers/feature_interaction/dot_interaction_test.py

```diff
@@ -8,10 +12,18 @@
 from keras_rs.src.layers.feature_interaction.dot_interaction import (
     DotInteraction,
 )
+from keras_rs.src.utils import tpu_test_utils
 
 
 class DotInteractionTest(testing.TestCase, parameterized.TestCase):
     def setUp(self):
+        super().setUp()
+        if keras.backend.backend() == "tensorflow":
+            tf.debugging.disable_traceback_filtering()
+
+        self.on_tpu = "TPU_NAME" in os.environ
+        self._strategy = tpu_test_utils.get_tpu_strategy(self)
```

Comment:
Let's not do that in each test class.

For `self.on_tpu`, you can probably do it once for all in `testing.TestCase.setUp`.

For `self.strategy`, let's make a  property that is lazily created, in `testing.TestCase.setUp`.

```python
@property
def strategy(self):
    if hasattr(self, "_strategy"):
        return self._strategy

    ...
    self._strategy = ...
    return self._strategy
```

Although, we may have to do that in a piece of code that is shared between all tests, like `conftest.py`.